bindings/python/test/test_arrow.py

```diff
@@ -244,27 +244,43 @@ def test_pymongo_error(self):
             exc.exception.details.keys(), {"nInserted", "writeConcernErrors", "writeErrors"}
         )
 
-    def _create_data(self):
+    def _create_data(self, block=False, nested=True):
```

Comment:
What does "block" mean?  It looks like this is only ever called with `block=True`.  When would we not want to include the extension types?